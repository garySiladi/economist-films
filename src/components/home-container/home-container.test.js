// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import HomeContainer from './home-container';

describe('HomeContainer: ', () => {
  test('renders correctly', () => {
    const tree : string = renderer.create(<HomeContainer />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('fetches correctly', () => {
    const elem = shallow(<HomeContainer />);
    expect(elem.state()).toMatchObject({ series: {} });
  });

  test('findSeriesByTitle returns correct object', () => {
    const result = HomeContainer.findSeriesByTitle('xxx', [{ title: 'xxx' }, { title: 'xzx' }]);
    expect(result).toMatchObject({ title: 'xxx' });
  });

  test('findSeriesByTitle returns null', () => {
    const result = HomeContainer.findSeriesByTitle('xxx', [{ title: 'xyx' }, { title: 'xzx' }]);
    expect(result).toBe(undefined);
  });
});
