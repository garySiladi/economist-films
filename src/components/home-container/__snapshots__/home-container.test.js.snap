exports[`HomeContainer:  renders correctly 1`] = `null`;
